netket/operator/_discrete_operator_jax.py

```diff
@@ -245,11 +246,9 @@ def to_qobj(self):  # -> "qutip.Qobj"
         )
 
     def __matmul__(self, other):
-        if (
-            isinstance(other, np.ndarray)
-            or isinstance(other, jnp.ndarray)
-            or isinstance(other, JAXSparse)
-        ):
+        if isinstance(other, JAXSparse):
```

Comment:
@wdphy16 the only questionable thing is that jax sparse matrices do not support multiplication by another jax CSR sparse matrix, so I had to ensure that when one tries to do `jaxoperator @ sparsevector` we convert it to dense first.

This breaks current behaviour where, as we use BCOO, we get a sparse vector out. 

But I think that's not important? Hopefully one never does this kind of multiplication...

Reply:
Ah it's a bit unexpected for me, but anyway JAX sparse matrices are experimental so we can make breaking changes

In real use cases, I guess we usually first do NetKet operator multiplications then convert it to a JAX sparse matrix